pkg/store/proxy.go

```diff
@@ -594,6 +597,16 @@ func storeMatches(ctx context.Context, s Client, mint, maxt int64, matchers ...*
 	if !LabelSetsMatch(matchers, extLset...) {
 		return false, fmt.Sprintf("external labels %v does not match request label matchers: %v", extLset, matchers)
 	}
+
+	for _, m := range matchers {
```

Comment:
Small suggestion: this will now add extra work all users of store/proxy.go. I suggest moving this whole loop into `s.MatchesMetricName()` instead.

So, it would be

`func MatchesMetricName(matches []labels.Matchers) bool`. The allow-all filter would just do no work at all and the cuckoo version would iterate through the given matchers.